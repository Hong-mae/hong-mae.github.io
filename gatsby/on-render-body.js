"use strict";

const React = require("react");
const siteConfig = require("../config.js");

// eslint-disable-next-line import/no-webpack-loader-syntax, import/no-unresolved
const katexStylesheet = require("!css-loader!../static/css/katex/katex.min.css");

const onRenderBody = (
  { setHeadComponents, setPostBodyComponents },
  pluginOptions
) => {
  const { useKatex } = siteConfig;

  if (useKatex) {
    setHeadComponents([
      React.createElement("style", {
        key: "katex-inline-stylesheet",
        dangerouslySetInnerHTML: { __html: katexStylesheet.toString() },
      }),
    ]);
  }

  setPostBodyComponents([
    <script
      key={"kakao_adfit_script"}
      type="text/javascript"
      async
      src="//t1.daumcdn.net/kas/static/ba.min.js"
    />,
  ]);
};

module.exports = onRenderBody;
