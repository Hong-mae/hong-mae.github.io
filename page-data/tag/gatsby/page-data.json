{"componentChunkName":"component---src-templates-tag-template-js","path":"/tag/gatsby","result":{"data":{"site":{"siteMetadata":{"title":"Moong bee's Dev.log","subtitle":"Moong bee의 개발 스터디 블로그."}},"allMarkdownRemark":{"edges":[{"node":{"fields":{"slug":"/posts/create-react-blog-with-gatsby-template-final","categorySlug":"/category/react/"},"frontmatter":{"title":"리액트 + Gatsby 를 이용한 블로그 만들기_final","date":"2020/10/08 12:07:24 GMT+0900","category":"React","description":"SEO를 지원하는 블로그를 위해 리액트 + Gatsby 이용하여 블로그를 만들자. final Gatsby에서 제공하는 plugins!"}}},{"node":{"fields":{"slug":"/posts/create-react-blog-with-gatsby-template-3","categorySlug":"/category/react/"},"frontmatter":{"title":"리액트 + Gatsby 를 이용한 블로그 만들기_#3","date":"2020/10/07 00:55:12 GMT+0900","category":"React","description":"SEO를 지원하는 블로그를 위해 리액트 + Gatsby 이용하여 블로그를 만들자. #3 Gatsby를 실행하는 방법!"}}},{"node":{"fields":{"slug":"/posts/create-react-blog-with-gatsby-template-2","categorySlug":"/category/react/"},"frontmatter":{"title":"리액트 + Gatsby 를 이용한 블로그 만들기_#2","date":"2020/10/04 00:22:23 GMT+0900","category":"React","description":"SEO를 지원하는 블로그를 위해 리액트 + Gatsby 이용하여 블로그를 만들자. #2 기본적인 Gatsby 사용법!"}}},{"node":{"fields":{"slug":"/posts/create-react-blog-with-gatsby-template-1","categorySlug":"/category/react/"},"frontmatter":{"title":"리액트 + Gatsby 를 이용한 블로그 만들기_#1","date":"2020/10/02 00:37:37 GMT+0900","category":"React","description":"SEO를 지원하는 블로그를 위해 리액트 + Gatsby 이용하여 블로그를 만들자. #1, Gatsby를 쓰는 이유!"}}}]}},"pageContext":{"tag":"Gatsby","currentPage":0,"postsLimit":4,"postsOffset":0,"prevPagePath":"/tag/gatsby","nextPagePath":"/tag/gatsby/page/1","hasPrevPage":false,"hasNextPage":true}},"staticQueryHashes":["251939775","401334301","825871152"]}