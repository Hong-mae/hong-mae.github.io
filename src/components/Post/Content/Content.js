// @flow strict
import React from 'react';
import { useSiteMetadata } from '../../../hooks';
import Adfit from '../../Adfit';
import styles from './Content.module.scss';

type Props = {
    body: string,
    title: string,
    date: string,
    description?: string,
};

const Content = ({ body, title, date, description }: Props) => {
    const { author } = useSiteMetadata();

    return (
        <div className={styles['content']}>
            <div className={styles['content__header']}>
                <h1 className={styles['content__title']}>{title}</h1>
                <div className={styles['content__description']}>
                    {description}
                </div>
                <div className={styles['content__author']}>
                    {author.name}
                    <span className={styles['content__bar']} />
                    {new Date(date).toLocaleDateString('ko-KR', {
                        year: 'numeric',
                        month: 'short',
                        day: 'numeric',
                    })}
                </div>
            </div>
            <Adfit
                name="content_adfit"
                pc="DAN-ecKceu8n4NKFPXSU"
                mobile="DAN-rFvENKyKGK9NwuNt"
            />
            <div
                className={styles['content__body']}
                dangerouslySetInnerHTML={{ __html: body }}
            />
        </div>
    );
};

export default Content;
