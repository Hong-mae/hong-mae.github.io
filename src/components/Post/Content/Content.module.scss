@import '../../../assets/scss/variables';
@import '../../../assets/scss/mixins';

.content {
  &__header {
    border-bottom: 1px solid $color-gray;
    @include padding(2.5, 0, 3.5);
    @include margin-bottom(2);
  }

  &__author {
    text-align: center;
    color: $color-gray;
    font-size: 12px;
  }

  &__bar {
    display: inline-block;
    width: 1px;
    height: 9px;
    margin: 0 5px;
    background-color: #ebebeb;
  }

  &__title {
    font-size: $typographic-base-font-size * 2.5;
    // max-width: $layout-post-width;
    // margin-left: auto;
    // margin-right: auto;
    font-weight: 600;
    text-align: center;
    @include line-height(1.65);
    @include margin-top(0);
    @include margin-bottom(0);
  }

  &__adfit {
    text-align: center;
  }

  &__description {
    text-align: center;
    font-size: $typographic-base-font-size;
  }

  &__body {
    & details {
      background-color: #00ff5a47;
      @include padding(0.65, 0.65);

      ul,
      ol {
        margin: 0;
      }
    }

    & figure {
      @include margin-bottom(1);

      & blockquote {
        font-style: italic;
        text-align: center;
        margin-top: 0;
        @include padding(1, 0);

        & p {
          max-width: $layout-post-width;
          font-size: $typographic-base-font-size * 1.6817;
          margin-top: 0;
          @include margin-bottom(1);
          @include line-height(1.5);
        }
      }
    }

    & a {
      text-decoration: underline;
    }

    // & * {
    // max-width: $layout-post-width;
    // margin-left: auto;
    // margin-right: auto;
    // }

    & img {
      max-width: 100%;
    }

    & table {
      border-collapse: collapse;
      margin-left: auto;
      margin-right: auto;

      th,
      td {
        border: 1px solid;
        padding: 5px 10px;

        ul,
        ol {
          padding: 0;
          list-style: none;
        }
      }
    }
  }
}

@include breakpoint-md {
  .content {
    &__title {
      font-size: $typographic-base-font-size * 3;
      @include line-height(2.25);
      @include margin-top(2.25);
      @include margin-bottom(0.25);
    }

    &__body {
      font-size: $typographic-base-font-size * 1.125;
      @include line-height(1.125);
      @include margin-bottom(1.125);

      & p {
        font-size: $typographic-base-font-size * 1.125;
        @include line-height(1.125);
        @include margin-bottom(1.125);
      }
    }
  }
}
