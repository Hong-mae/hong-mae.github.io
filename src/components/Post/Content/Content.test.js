// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import { useStaticQuery, StaticQuery } from 'gatsby';
import Content from './Content';
import siteMetadata from '../../../../jest/__fixtures__/site-metadata';
import type { RenderCallback } from '../../../types';

describe('Content', () => {
  const props = {
    ...siteMetadata,
    title: 'test',
    body: '<p>test</p>',
    date: '2021/09/09 18:00:00 GMT+0900',
  };

  beforeEach(() => {
    StaticQuery.mockImplementationOnce(
      ({ render }: RenderCallback) => render(props),
      useStaticQuery.mockReturnValue(props)
    );
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Content {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
