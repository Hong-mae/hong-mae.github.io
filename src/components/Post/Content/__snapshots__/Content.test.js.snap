// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content renders correctly 1`] = `
<div
  className="content"
>
  <div
    className="content__header"
  >
    <h1
      className="content__title"
    >
      test
    </h1>
    <div
      className="content__description"
    />
    <div
      className="content__author"
    >
      Test name
      <span
        className="content__bar"
      />
      2021년 9월 9일
    </div>
  </div>
  <div
    className="adfit content_adfit"
  >
    <div
      className="ad_area"
    />
  </div>
  <div
    className="content__body"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>test</p>",
      }
    }
  />
</div>
`;
